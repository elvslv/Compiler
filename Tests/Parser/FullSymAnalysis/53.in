var i, j, n: integer;
    s, tmp: real;
procedure rec(num, l: integer; s: real);
var j: integer;
    s1: real;
begin
  for j := num to l  do
  begin
    if (l - j >= num) and (s - 3 <= j) and (l >= num)then
    begin
      rec(num, l - j, s + 5);
    end;
    if (l - j = 0) and (j >= num) and (s - 4.5 <= j) and (l >= num)then
      s1 := 1e-10;
  end; 
end;
begin
  for i := 1 to n div 2 + 1 do
    rec(i, n - i, i);
end.