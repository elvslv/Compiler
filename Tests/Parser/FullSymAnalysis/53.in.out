var I : INTEGER;
var J : INTEGER;
var N : INTEGER;
procedure REC(var L : INTEGER;var NUM : INTEGER;var S : REAL;);
begin                                
|                                    
|___for                              
|   |                                
|   |___:=                           
|   |   |                            
|   |   |___J                        
|   |   |                            
|   |   |___NUM                      
|   |                                
|   to                               
|   |                                
|   |___L                            
|   |                                
|   do                               
|   |                                
|   |___begin                        
|       |                            
|       |___if                       
|       |   |                        
|       |   |___AND                  
|       |   |   |                    
|       |   |   |___AND              
|       |   |   |   |                
|       |   |   |   |___>=           
|       |   |   |   |   |            
|       |   |   |   |   |___-        
|       |   |   |   |   |   |        
|       |   |   |   |   |   |___L    
|       |   |   |   |   |   |        
|       |   |   |   |   |   |___J    
|       |   |   |   |   |            
|       |   |   |   |   |___NUM      
|       |   |   |   |                
|       |   |   |   |___<=           
|       |   |   |       |            
|       |   |   |       |___-        
|       |   |   |       |   |        
|       |   |   |       |   |___S    
|       |   |   |       |   |        
|       |   |   |       |   |___ItoR 
|       |   |   |       |       |    
|       |   |   |       |       |___3
|       |   |   |       |            
|       |   |   |       |___ItoR     
|       |   |   |           |        
|       |   |   |           |___J    
|       |   |   |                    
|       |   |   |___>=               
|       |   |       |                
|       |   |       |___L            
|       |   |       |                
|       |   |       |___NUM          
|       |   |                        
|       |   then                     
|       |   |                        
|       |   |___begin                
|       |       |                    
|       |       |___()               
|       |       |   |                
|       |       |   |___REC          
|       |       |   |                
|       |       |   |___NUM          
|       |       |   |                
|       |       |   |___-            
|       |       |   |   |            
|       |       |   |   |___L        
|       |       |   |   |            
|       |       |   |   |___J        
|       |       |   |                
|       |       |   |___+            
|       |       |       |            
|       |       |       |___S        
|       |       |       |            
|       |       |       |___ItoR     
|       |       |           |        
|       |       |           |___5    
|       |       |                    
|       |       end                  
|       |                            
|       |___if                       
|       |   |                        
|       |   |___AND                  
|       |   |   |                    
|       |   |   |___AND              
|       |   |   |   |                
|       |   |   |   |___AND          
|       |   |   |   |   |            
|       |   |   |   |   |___=        
|       |   |   |   |   |   |        
|       |   |   |   |   |   |___-    
|       |   |   |   |   |   |   |    
|       |   |   |   |   |   |   |___L
|       |   |   |   |   |   |   |    
|       |   |   |   |   |   |   |___J
|       |   |   |   |   |   |        
|       |   |   |   |   |   |___0    
|       |   |   |   |   |            
|       |   |   |   |   |___>=       
|       |   |   |   |       |        
|       |   |   |   |       |___J    
|       |   |   |   |       |        
|       |   |   |   |       |___NUM  
|       |   |   |   |                
|       |   |   |   |___<=           
|       |   |   |       |            
|       |   |   |       |___-        
|       |   |   |       |   |        
|       |   |   |       |   |___S    
|       |   |   |       |   |        
|       |   |   |       |   |___4.5  
|       |   |   |       |            
|       |   |   |       |___ItoR     
|       |   |   |           |        
|       |   |   |           |___J    
|       |   |   |                    
|       |   |   |___>=               
|       |   |       |                
|       |   |       |___L            
|       |   |       |                
|       |   |       |___NUM          
|       |   |                        
|       |   then                     
|       |   |                        
|       |   |___begin                
|       |       |                    
|       |       |___:=               
|       |       |   |                
|       |       |   |___S1           
|       |       |   |                
|       |       |   |___1E-10        
|       |       |                    
|       |       end                  
|       |                            
|       end                          
|                                    
end                                  

var S : REAL;
var TMP : REAL;

begin                                
|                                    
|___for                              
|   |                                
|   |___:=                           
|   |   |                            
|   |   |___I                        
|   |   |                            
|   |   |___1                        
|   |                                
|   to                               
|   |                                
|   |___+                            
|   |   |                            
|   |   |___DIV                      
|   |   |   |                        
|   |   |   |___N                    
|   |   |   |                        
|   |   |   |___2                    
|   |   |                            
|   |   |___1                        
|   |                                
|   do                               
|   |                                
|   |___begin                        
|       |                            
|       |___()                       
|       |   |                        
|       |   |___REC                  
|       |   |                        
|       |   |___I                    
|       |   |                        
|       |   |___-                    
|       |   |   |                    
|       |   |   |___N                
|       |   |   |                    
|       |   |   |___I                
|       |   |                        
|       |   |___ItoR                 
|       |       |                    
|       |       |___I                
|       |                            
|       end                          
|                                    
end                                  
